src/pyk/proof/reachability.py

```diff
@@ -62,17 +83,28 @@ def status(self) -> ProofStatus:
     def from_dict(cls: type[APRProof], dct: Mapping[str, Any], proof_dir: Path | None = None) -> APRProof:
         cfg = KCFG.from_dict(dct['cfg'])
         id = dct['id']
+        if len(dct['dependencies']) > 0:
+            if proof_dir is None:
+                raise ValueError('The serialized proof has dependencies but no proof_dir was specified')
+            dependencies = [APRProof.read_proof(id, proof_dir=proof_dir) for id in dct['dependencies']]
+        circularity = dct['circularity']
```

Comment:
```suggestion
        dependencies = [APRProof.read_proof(id, proof_dir=proof_dir) for id in dct['dependencies']]
        circularity = dct['circularity']
```

I guess we'll get a good stack trace if this fails anyway.

Reply:
@ehildenb the type checker won't allow this, since `proof_dir` can be `None` in the caller, but cannot be `None` in `APRProof.read_proof`.